cmd/deploy.go

```diff
@@ -139,43 +139,63 @@ func deploy(environmentType string, serviceName string, serviceType string, fold
 	}
 	s1.Stop()
 
-	s2 := spinner.New(spinner.CharSets[26], 100*time.Millisecond)
-
 	//staging the deploy, total time is about 2100
-	progressBar("Initializing deployment... ", 700)
-	fmt.Print("\n")
-	progressBar("Deploying service ...      ", 700)
-	fmt.Print("\n")
-	progressBar("Finalizing deployment...   ", 700)
-	fmt.Print("\n")
+	// progressBar("Initializing deployment... ", 700)
+	// fmt.Print("\n")
+	// progressBar("Deploying service ...      ", 700)
+	// fmt.Print("\n")
+	// progressBar("Finalizing deployment...   ", 700)
+	// fmt.Print("\n")
 
 	servUrl := ""
-	s2.Prefix = "Fetching service URL... "
+
+	p := mpb.New(mpb.WithWidth(64))
+	defer p.Wait()
+	total := 68
+
+	bar := getProgressBar(p, "Deploying service ...", total)
 
 	for {
 		update, err := stream.Recv()
 		if err == io.EOF {
 			break
 		} else if err != nil {
+			// bar.Abort(true)
+			//s2.Stop()
 			log.Fatalf("server side error: %s", err.Error())
 		}
 
-		if msg := update.GetDeploymentUpdate(); msg != nil {
+		// bar.Increment()
+		deployUpdate := update.GetDeploymentUpdate()
+		if deployUpdate != nil {
+			bar.IncrBy(2) // there are 34 msgs the server send to the client, so total is 68/34 = 2 per message will get us to 100
```

Comment:
How did you get the 34 number? Did you just count it manually? Won't this be variable based on how long it takes to actually deploy something?